packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -97,6 +104,40 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
             new UriAwareCommandHandler(this.selectionService, {
                 execute: (uri: URI) => this.saveAs(uri),
             }));
+        commands.registerCommand(WorkspaceCommands.UPLOAD, new FileSelection.CommandHandler(this.selectionService, {
+            multi: false,
+            isEnabled: selection => this.isEnabled(selection),
+            isVisible: selection => this.isVisible(selection),
+            execute: selection => this.upload(selection)
+        }));
+    }
+
+    protected isEnabled(selection: FileSelection | undefined): boolean {
+        return !environment.electron.is();
```

Comment:
What happens if we have a multi-root workspace without any roots?

Reply:
I was also considering this issue, but I'm just thinking how is this a possibility that when theia opens up there is no workspace present?

Reply:
It is definitely possible for the application to have no workspace present, and for it to be in multi-root state without any actual roots.

Reply:
No workspace present is equivalent of starting without a workspace, or executing the command `File` > `Close Workspace`

Reply:
This use case feels a bit odd to me:

<div align='center'>


![b](https://user-images.githubusercontent.com/40359487/64260738-719ffe00-cef9-11e9-9043-21075aa1c82d.gif)


</div>

**How to test:**
1. open a workspace
2. add another folder using `Add Folder to Workspace...`
3. remove both roots using `Remove Folder from Workspace...`
4. the result is a workspace without roots
5. attempt to `Upload Files...`
6. user is prompted to choose a file, when selected nothing happens

Reply:
Just tested what happens after choosing "Close Workspace".

The menu item "upload files" is enabled, but nothing will happen if we press it, though it will throw error in the background:

![image](https://user-images.githubusercontent.com/29387772/64260964-be83d480-cef9-11e9-8b0c-4039d3185824.png)

Reply:
Something like `File: Upload Files...` should at least be only visible if a workspace is currently opened I believe. It won't make sense to display it in this scenario.